hotrod/src/main/java/io/hyperfoil/hotrod/steps/HotRodRequestBuilder.java

```diff
@@ -0,0 +1,59 @@
+package io.hyperfoil.hotrod.steps;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.kohsuke.MetaInfServices;
+
+import io.hyperfoil.api.config.Name;
+import io.hyperfoil.api.config.Step;
+import io.hyperfoil.api.config.StepBuilder;
+import io.hyperfoil.core.builders.BaseStepBuilder;
+import io.hyperfoil.core.session.SessionFactory;
+import io.hyperfoil.hotrod.api.HotRodOperation;
+import io.hyperfoil.hotrod.resource.HotRodResourceKey;
+
+/**
+ * Issues a HotRod request and registers handlers for the response.
+ */
+@MetaInfServices(StepBuilder.class)
+@Name("hotrodRequest")
+public class HotRodRequestBuilder extends BaseStepBuilder<HotRodRequestBuilder> {
+
+   private HotRodOperationBuilder operation;
+   private HotRodCacheNameBuilder cacheName;
+   private int stepId = 1;
+
+   @Override
+   public List<Step> build() {
+
+      HotRodResourceKey key = new HotRodResourceKey();
+      HotRodRequestStep step = new HotRodRequestStep(stepId, key, operation.build(), cacheName.build(),
+            SessionFactory.access("cacheKey"), SessionFactory.access("cacheValue"));
```

Comment:
Don't hardcode variables for key and value. To align it with other similar properites in HF you should support these forms:
```yaml
key:
  value: foo
```
```yaml
key:
  fromVar: bar
```
```yaml
key:
  pattern: foo${bar}
```
```yaml
key: foo${bar} #inline pattern
```
Check out `StringGeneratorImplBuilder` (though you don't want to limit it to strings). The pattern (both inline and explicit) notation is optional, but `fromVar` and `value` (constant string) should be there.

Reply:
A benchmark can write to multiple entries in different caches. You might want to execute those operations in parallel etc... When you do
```
- randomInt: 1 .. 100 -> someKey
- cachePut:
    key: ${someKey}
    value: ...
`
it's obvious that you've generated something and then you somehow use it.

Reply:
Can I have the scenario like?
```
scenario:
  - example:
      - randomInt: 1 .. 100 -> someKey
      - randomUUID: cacheValue
      - hotrodRequest:
          put: my-cache
          key: key-${someKey}-some-suffix
          value: ${cacheValue}
```

Reply:
Sure. However when you add the `get` operation you'll end up with
```yaml
- hotrodRequest:
    get: my-cache
    key: ...
    value: ??????
```
This construct is 'syntactically' correct (passes schema) but you have the throw an exception when the user provides a value for `get`. Then you add `putAll` with `keys` and in the end every operation will have to check presence of unrelated properties. That's why it would make sense to scope the properties for each operation, e.g.
```yaml
- hotrodRequest:
    cache: my-cache
    put:
      key: ...
      value: ...
```

Reply:
Fixed with: Key and Value Generator